src/awkward/_v2/contents/indexedoptionarray.py

```diff
@@ -341,9 +602,9 @@ def _sort_next(
         )
 
         next_length = index_length - numnull[0]
-        nextparents = ak._v2.index.Index64.zeros(next_length, nplike)
-        nextcarry = ak._v2.index.Index64.zeros(next_length, nplike)
-        outindex = ak._v2.index.Index64.zeros(index_length, nplike)
+        nextparents = ak._v2.index.Index64.empty(next_length, nplike)
+        nextcarry = ak._v2.index.Index64.empty(next_length, nplike)
+        outindex = ak._v2.index.Index64.empty(index_length, nplike)
```

Comment:
This and the above is a good catch. I also use `zeros` (or more often `ones` or `full` with value `123`) for debugging, but then we want to switch over to `empty` for production code.